google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/GooglePubSubSourceGrpc.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.pubsub
+
+import akka.stream.{Attributes, Materializer, Outlet, SourceShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+
+import scala.util.{Failure, Success, Try}
+import GooglePubSubSourceGrpc._
+import com.google.pubsub.v1
+
+import scala.collection.JavaConverters._
+import scala.collection.immutable
+
+@akka.annotation.InternalApi
+private final class GooglePubSubSourceGrpc(parallelism: Int, grpcApi: GrpcApi)
+    extends GraphStage[SourceShape[v1.ReceivedMessage]] {
+
+  val out: Outlet[v1.ReceivedMessage] = Outlet("GooglePubSubSourceGrpc.out")
+  override val shape: SourceShape[v1.ReceivedMessage] = SourceShape(out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with OutHandler {
+      private var state: State = Pending
+      private var fetches: Int = 0
+
+      def fetch(implicit mat: Materializer): Unit = {
```

Comment:
Maybe I don't understand that but what is the point of passing Materializer here if it can be retrieved inside GraphStageLogic in same way as you do in onPull?

Reply:
I understand it is async, but I think code running inside async callback has access to materializer.